content/Projects/Observability/kube-rbac-proxy.md

```diff
@@ -0,0 +1,106 @@
+
+- **Static authorization**: Static authorization involves configuring `kube-rbac-proxy` to allow access to certain resources or non-resources which are evaluated against the `Role` or `ClusterRole` RBAC permissions the user or the service account has. The example below demonstrates how this can be employed to give access to a known `ServiceAccount` to the `/metrics` endpoint. `/metrics` endpoints exposed by various monitoring components are protected this way. Note that after the initial user or service account authentication, the request is matched against a comma-separated list of paths, as defined by the [`--allow-path`](https://github.com/brancz/kube-rbac-proxy/blob/067e14a2d1ecdfe8c18da6b0a0507cd4684e2c1c/cmd/kube-rbac-proxy/app/options/options.go#L83) flag, [like so](https://github.com/openshift/cluster-monitoring-operator/blob/4e8efd9864bff3ff46499e86fef8bba1e0178f54/assets/alertmanager/alertmanager.yaml#L96).
+
+```yaml
+apiVersion: v1
+kind: Secret
+...
+stringData:
+  # "path" is the path to match against the request path.
+  # "resourceRequest" is a boolean indicating whether the request is for a resource or not.
+  # "user" is the user to match against the request user.
+  # "verb" is the verb to match against the corresponding request RBAC verb.
+  config.yaml: |-
+    "authorization":
+      "static":
+      - "path": "/metrics"
+        "resourceRequest": false
+        "user":
+          "name": "system:serviceaccount:openshift-monitoring:prometheus-k8s"
+        "verb": "get"
+```
+
+On the other hand, the example below depicts restricted access to a resource, i.e., `monitoring.coreos.com/prometheusrules` in the `openshift-monitoring` namespace. Note that all applicable omitted configuration settings are interpreted as wildcards.
+
+```yaml
+apiVersion: v1
+kind: Secret
+...
+stringData:
+  # "resourceAttributes" describes attributes available for resource request authorization.
+  # "rewrites" describes how SubjectAccessReview may be rewritten on a given request.
+  # "rewrites.byQueryParameter" describes which HTTP URL query parameter is to be used to rewrite a SubjectAccessReview 
+  # on a given request.
+  config.yaml: |-
+    "authorization":
+      "resourceAttributes":
+        "apiGroup": "monitoring.coreos.com"
+        "namespace": "openshift-monitoring"
+        "resource": "prometheusrules"
+      "rewrites":
+        "byQueryParameter":
+          "name": "namespace"
+```
+
+For more details, refer to the `kube-rbac-proxy`'s [static authorization](https://github.com/brancz/kube-rbac-proxy/blob/4a44b610cd12c4cfe076a2b306283d0598c1bb7a/examples/static-auth/README.md#L169) example.
```

Comment:
the upstream example doesn't match what we describe before? e.g. it's static auth while the use case is auth with attribute rewrites.